runtime-modules/bounty/src/lib.rs

```diff
@@ -194,19 +217,35 @@ pub enum BountyStage {
 #[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
 pub enum BountyMilestone<BlockNumber> {
     /// Bounty was created at given block number.
-    Created(BlockNumber),
+    /// Boolean value defines whether the bounty has some funding contributions.
+    Created {
+        /// Bounty creation block.
+        created_at: BlockNumber,
+        /// Bounty has already some contributions.
+        has_contributions: bool,
+    },
 
     /// A bounty was canceled.
     Canceled,
 
-    /// A bounty funding was successful on the provided block.
-    /// The stage is set when the funding exceeded max funding amount.
-    MaxFundingReached(BlockNumber),
+    /// Creator funds (initial funding and/or cherry) were withdrawn.
+    CreatorFundsWithdrawn,
```

Comment:
Why is there a distinct extrinsic and thus milestone for withdrawing funding? All there is is bounty cancellation? 

See design doc
> The creator can cancel at any time before this, something that is of particular importance of in perpetual funding periods. If no one has contributed, any cherry is returned and the bounty is instantl removed, but if at least one contributor has funded, then the cherry will be split amont all contributors, and the bounty goes to the withdraw period.

So cancelling a bounty covers possible recovery of bounty contributed, if relevant. Right now you have two separate extrinsics and milestones. I am not actually sure you even need two `Canceled` and `CreatorFundsWithdrawn` even if we wanted two separate extrinsics, as the runtime should do the exact same thing going forward in both cases?